### PR TITLE
Eliminate boxing allocation

### PR DESCRIPTION
**Pull request by drewnoakes**
## Proposed changes

This change removes 537,404 bytes of boxed enum allocations while loading GE with the gitextensions repo.

## Test methodology 

- Manual testing
 
----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/UserControls/RevisionGrid/RevisionGridControl.cs [hunk 1]
```diff
@@ -884,9 +884,11 @@ public void ForceRefreshRevisions()
 
                 SelectInitialRevision(newCurrentCheckout);
 
-                // NB: Build ref filter before updating the property to reduce the number of chage events.
+                const RefFilterOptions gitNotesOptions = RefFilterOptions.All | RefFilterOptions.Boundary;
+
+                // NB: Build ref filter before updating the property to reduce the number of change events.
                 RefFilterOptions refFilterOptions = RefFilterOptions;
-                if (!AppSettings.ShowGitNotes && refFilterOptions.HasFlag(RefFilterOptions.All | RefFilterOptions.Boundary))
+                if (!AppSettings.ShowGitNotes && (refFilterOptions & gitNotesOptions) == gitNotesOptions)
```
Comment (RussKie) [inline GitUI/UserControls/RevisionGrid/RevisionGridControl.cs:891]: IIRC .NET Core has resolved this allocations, hasn't it? https://github.com/gitextensions/gitextensions/pull/8522 makes this change obsolete.

Comment (drewnoakes) [reply GitUI/UserControls/RevisionGrid/RevisionGridControl.cs:891]: Yes, it's not needed under .NET Core. What is the timeline for porting?

Until such a time, I suggest merging this. I ran a trace over GE startup and this was a quick win from the top 30 types allocated on the heap.

Comment (gerhardol) [reply GitUI/UserControls/RevisionGrid/RevisionGridControl.cs:891]: then this should go into 3.4 too

**Comment by RussKie** [comment]
Why 3.4?

**Comment by gerhardol** [comment]
> Why 3.4?

Typo, edited to 3.5

**Comment by RussKie** [comment]
Works for me

File: GitUI/UserControls/RevisionGrid/RevisionGridControl.cs (hunk 1 above)
Comment (drewnoakes) [reply GitUI/UserControls/RevisionGrid/RevisionGridControl.cs:891]: Sibling PR targeting 3.5: #8889

**Review by gerhardol — COMMENTED** [review]
+1

**Review by RussKie — APPROVED** [review]

**Pull request merged**